aten/src/ATen/core/LegacyTypeDispatch.h

```diff
@@ -62,4 +62,10 @@ struct CAFFE2_API AutoNonVariableTypeMode {
   c10::impl::ExcludeTensorTypeIdGuard guard_;
 };
 
+namespace impl {
+inline bool variable_is_excluded() {
```

Comment:
I feel that the function name is a bit unintuitive (it wasn't immediately clear to me at first sight what it is checking). Perhaps we could name it something like `is_non_variable_type` ? (not sure if it is better)

Reply:
The trouble is, "what", precisely, is the non variable type? Not sure there is much of an improvement here either, alas.

Reply:
Does this only effect dispatch now?  If that's true, "variable_excluded_from_dispatch" is probably clearer.

Reply:
Yes, it only excludes dispatch. I suppose I like the new name.